.github/workflows/maintainer-approval.yml

```diff
@@ -1,7 +1,8 @@
 name: Maintainer approval
 
 on:
-  pull_request:
+  pull_request_review:
+    types: [submitted]
```

Comment:
This approach doesn't work if a new commit removes the existing check check result.